test/services/post-install-utility.test.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-console */
+const tmp = require('tmp');
+const chalk = require('chalk');
+const { expect, test, constants } = require('@twilio/cli-test');
+const sinon = require('sinon');
+const { ConfigData } = require('@twilio/cli-core').services.config;
+
+const { PostInstallDisplayManager, PORT_WARNING } = require('../../src/services/post-install-utility');
+
+describe('services', () => {
+  describe('post-install-utility', () => {
+    beforeEach(() => {
+      tempConfigDir = tmp.dirSync({ unsafeCleanup: true });
+      warnSpy = sinon.spy(console, 'warn');
+      logSpy = sinon.spy(console, 'log');
+    });
+
+    afterEach(() => {
+      tempConfigDir.removeCallback();
+      warnSpy.restore();
+      logSpy.restore();
+    });
+
+    test.it('should display grid when no profiles or projects configured', () => {
+      const configData = new ConfigData();
+      const displayManager = new PostInstallDisplayManager(tempConfigDir.name, configData);
+
+      displayManager.displayMessage();
+      expect(displayManager.userConfig).to.not.be.undefined;
+      expect(console.log.calledWith('* To get started, please create a twilio-cli profile:                   *')).to.be
+        .true;
+      expect(console.log.calledWith('*     twilio profiles:create                                            *')).to.be
+        .true;
+      expect(console.log.calledWith('*     OR                                                                *')).to.be
+        .true;
+      expect(console.log.calledWith('*     twilio login                                                      *')).to.be
+        .true;
+    });
+
+    test.it('should display port warning if projects set', () => {
+      const configData = new ConfigData();
+      configData.addProject('DUMMY_PROJECT', constants.FAKE_ACCOUNT_SID);
+      const displayManager = new PostInstallDisplayManager(tempConfigDir.name, configData);
+
+      displayManager.displayMessage();
+      expect(console.warn.calledOnce).to.be.true;
+      expect(console.warn.calledWith(chalk.yellowBright(` » ${PORT_WARNING}`))).to.be.true;
```

Comment:
Again, shouldn't be so specific. Would also recommend `PORT_WARNING` not be exported and instead just do assertions on part of the text like `twilio profiles:port` is present or not.

Reply:
Yeah, Totally agree! 

Just couldn't find any `sinon` assertion which'd check **contains** relation of spy object param, v/s matching the exact param passed to the spy. Also- Couldn't test the assertions getting tunnelled to `stdout` instead. 
Could you please help me suggest a better way to test such param-contains relations? It might be totally something silly here that I might be missing here!

Reply:
Perhaps using a matcher: https://sinonjs.org/releases/latest/matchers/

```
expect(console.warn).to.have.been.calledWith(sinon.match(/twilio profiles:port/));
```

Reply:
Thank You! 💯